git-link.el

```diff
@@ -105,7 +105,18 @@
 
 ;; Matches traditional URL and scp style
 ;; This probably wont work for git remotes that aren't services
-(defconst git-link-remote-regex "\\([-.[:word:]]+\\)[:/]\\([^/]+/[^/]+?\\)\\(?:\\.git\\)?$")
+(defconst git-link-remote-regex
+  (rx
+   (submatch
+    (one-or-more (in "-" "." word digit)))
+   (any ":" "/")
+   (submatch
+    (one-or-more
+     (one-or-more (not (in "/")))
+     "/"
+     (+? (not (in "/")))))
+   (? ".git")
+   eol))
```

Comment:
Hi, thanks for this but let's use `"\\([-.[:word:]]+\\(?::[0-9]+\\)?\\)/\\([^/]+/[^/]+?\\)\\(?:\\.git\\)?$"` instead. No need to bring in `rx`.